tests/test_validation/steps/test_dataset_validator.py

```diff
@@ -0,0 +1,245 @@
+    )
+    with pytest.raises(ValidationError) as e:
+        dataset_validator.validate_dataset(
+            test_data.FIXED_DATE_INVALID_DS,
+            "DATE",
+            None,
+            None,
+            "FIXED",
+        )
+    assert e.value.errors == [
+        "Invalid value in #2 column for row with identifier: 4"
+    ]
+
+
+def test_temporality_fixed():
+    dataset_validator.validate_dataset(
+        test_data.FIXED_VALID_DS,
+        "STRING",
+        None,
+        None,
+        "FIXED",
+    )
+    with pytest.raises(ValidationError) as e:
+        dataset_validator.validate_dataset(
+            test_data.FIXED_INVALID_START_DS,
+            "STRING",
+            None,
+            None,
+            "FIXED",
+        )
+    assert e.value.errors == [
+        f"Invalid #3 and/or #4 columns for row with identifier: {id}"
+        for id in range(1, 5)
+    ]
+    with pytest.raises(ValidationError) as e:
+        dataset_validator.validate_dataset(
+            test_data.FIXED_INVALID_DUPLICATES_DS,
+            "STRING",
+            None,
+            None,
+            "FIXED",
+        )
+    assert e.value.errors == ["Duplicate identifiers in #1 column"]
+
+
+def test_temporality_status():
+    dataset_validator.validate_dataset(
+        test_data.STATUS_VALID_DS,
+        "STRING",
+        None,
+        None,
+        "STATUS",
+    )
+    with pytest.raises(ValidationError) as e:
+        dataset_validator.validate_dataset(
+            test_data.STATUS_INVALID_START_STOP_DS,
+            "STRING",
+            None,
+            None,
+            "STATUS",
+        )
+    assert e.value.errors == [
+        f"#3 column not equal to #4 column for row with identifier: {id}"
+        for id in range(1, 5)
+    ]
+    with pytest.raises(ValidationError) as e:
+        dataset_validator.validate_dataset(
+            test_data.STATUS_INVALID_DUPLICATES_DS,
+            "STRING",
+            None,
+            None,
+            "STATUS",
+        )
+    assert e.value.errors == [
+        "Same unit_id (#1 Column) has duplicate dates (#3 and #4 column)"
+    ]
+
+
+def test_temporality_event():
+    dataset_validator.validate_dataset(
+        test_data.EVENT_VALID_DS,
+        "STRING",
+        None,
+        None,
+        "EVENT",
+    )
+    with pytest.raises(ValidationError) as e:
+        dataset_validator.validate_dataset(
+            test_data.EVENT_INVALID_START_DS,
+            "STRING",
+            None,
+            None,
+            "EVENT",
+        )
+    assert e.value.errors == [
+        "Invalid #3 and/or #4 columns for row with identifier: 1"
+    ]
+    with pytest.raises(ValidationError) as e:
+        dataset_validator.validate_dataset(
+            test_data.EVENT_INVALID_TIMESPANS_DS,
+            "STRING",
+            None,
+            None,
+            "EVENT",
+        )
+    assert e.value.errors == [
+        'Invalid overlapping timespans for identifier "1"'
```

Comment:
Be sure to blacken the code

Reply:
black uses single qoutes around strings that contain double qoutes rather than \ escaping 👍🏻